src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -1,8 +1,156 @@
+
+    public Elevator() {
+
+        elevatorPositions.put(ElevatorHeight.HOME, ElevatorCal.POSITION_HOME_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L4, ElevatorCal.POSITION_SCORE_L4_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L3, ElevatorCal.POSITION_SCORE_L3_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L2, ElevatorCal.POSITION_SCORE_L2_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L1, ElevatorCal.POSITION_SCORE_L1_INCHES);
+        elevatorPositions.put(ElevatorHeight.SHALLOW_PREP, ElevatorCal.POSITION_PRE_SHALLOWCLIMB_INCHES);
+        elevatorPositions.put(ElevatorHeight.SHALLOW_CLIMB, ElevatorCal.POSITION_SCORE_SHALLOWCLIMB_INCHES);
+        initTalons();
+    }
+
+    private void initTalons() {
+        TalonFXConfigurator cfgLeft = leftMotor.getConfigurator();
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+
+        toApply.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        toApply.CurrentLimits.SupplyCurrentLimit = ElevatorCal.ELEVATOR_MOTOR_SUPPLY_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.StatorCurrentLimit = ElevatorCal.ELEVATOR_MOTOR_STATOR_SUPPLY_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.StatorCurrentLimitEnable = true;
+        // Slot 0 is for Scoring PID values and Slot 1 is for Climbing PID values
+        toApply.Slot0.kP = ElevatorCal.ELEVATOR_MOTOR_P;
+        toApply.Slot0.kI = ElevatorCal.ELEVATOR_MOTOR_I;
+        toApply.Slot0.kD = ElevatorCal.ELEVATOR_MOTOR_D;
+        toApply.Slot0.kV = ElevatorCal.ELEVATOR_MOTOR_FF;
+        toApply.Slot1.kP = ElevatorCal.ELEVATOR_CLIMB_P;
+        toApply.Slot1.kI = ElevatorCal.ELEVATOR_CLIMB_I;
+        toApply.Slot1.kD = ElevatorCal.ELEVATOR_CLIMB_D;
+        toApply.Slot1.kV = ElevatorCal.ELEVATOR_CLIMB_FF;
+        cfgLeft.apply(toApply);
+        Follower master = new Follower(leftMotor.getDeviceID(), true);
+        rightMotor.setControl(master);
+
+    }
+
+    public void setDesiredPosition(ElevatorHeight height) {
+        desiredPosition = height;
+    }
+
+    public void setControlParams(boolean isScoring) {
+        this.isScoring = isScoring;
+        currentSlotValue = isScoring?0:1; // 0 for scoring, 1 for climbing
+
+    }
+
+    private void controlPosition(double inputPositionInch) {
+        final PositionVoltage m_request = new PositionVoltage(0.0).withSlot(currentSlotValue);
+        double rotations = inputPositionInch / ElevatorConstants.DRUM_CIRCUMFEREMCE * ElevatorConstants.MOTOR_TO_DRUM_RATIO;
+        m_goal = new TrapezoidProfile.State(rotations, 0.0);
+
+        m_setpoint = (isScoring?m_profile_scoring:m_profile_climbing).calculate(0.02, m_setpoint, m_goal);
+
+        m_request.Position = m_setpoint.position;
+        m_request.Velocity = m_setpoint.velocity;
+
+        leftMotor.setControl(m_request);
+    }
+
+    public boolean atDesiredPosition(){
```

Comment:
just realized we might want at atPosition() method which checks for any position